pkg/lease/manager.go

```diff
@@ -0,0 +1,205 @@
+package lease
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	coordv1 "k8s.io/api/coordination/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Manager interface {
+	CreateOrGetLease(ctx context.Context, node *corev1.Node, duration time.Duration, holderIdentity string, namespace string) (*coordv1.Lease, bool, error)
+	UpdateLease(ctx context.Context, node *corev1.Node, lease *coordv1.Lease, currentTime *metav1.MicroTime, leaseDuration, leaseDeadline time.Duration, holderIdentity string) (bool, error)
```

Comment:
I don't remember why we had the `leaseDeadline` arg... doesn't make much sense to me. Isn't leaseDuration good enough? Just always update the lease with that duration? I don't see a usecase for "only update lease if it expires before deadline". WDYT?

Reply:
ping

Reply:
I think it's meant to allow more flexibility.

For example we can specify we'd like to reset the lease to expire 30 minutes  from now (leaseDuration) in case it's going to expire in the next hour (dealine).

In case we only use leaseDuration I don't think this sort of request can be accomplished.

Reply:
> reset the lease to expire 30 minutes from now (leaseDuration) in case it's going to expire in the next hour (dealine)

yes, but as I wrote in my comment above already, what's the usecase? Why would someone want to set lease `duration` depending on a `deadline`? Either I need the lease for `duration` or not... 🤔 IMHO this adding unneeded complexity to the API and to the implementation...